Library/Homebrew/tap.rb

```diff
@@ -210,12 +212,13 @@ def install(options = {})
     full_clone = options.fetch(:full_clone, false)
     quiet = options.fetch(:quiet, false)
     requested_remote = options[:clone_target] || default_remote
+    force_auto_update = options.fetch(:force_auto_update, nil)
```

Comment:
Instead of `fetch`ing with default as `nil`, maybe default to `false` like the `full_clone` variable above?

Reply:
The idea here was that, if `:force_auto_update` isn't set, we don't specifically mark the repository either way.

Reply:
To elaborate some more, I did it this way to handle the retapping. By having it default to `nil`, you don't need to specify the same value; it'll just use whatever is already in place. Additionally, it allows us to know if we need to mark/unmark the repo during a retap by checking for `nil`.

Reply:
It's possible that I'm misunderstanding, but this *should* have the same effect:

```ruby
force_auto_update = options.fetch(:force_auto_update, false)

# [... snip ...]

config["forceautoupdate"] = force_auto_update unless force_auto_update
```

In either case, it'd be good to add a comment stating the goal of a line like that (i.e., avoiding marking the repository either way).

Reply:
I'm pretty sure that wouldn't work.

Scenarios:

`options[:force_auto_update]` is `true`:
- `force_auto_update` is set to `true`
- `unless force_auto_update` fails:
  - `config["forceautoupdate"] = force_auto_update` is not executed
- the repo is not marked (**undesired outcome**)

`options[:force_auto_update]` is `false`:
- `force_auto_update` is set to `false`
- `unless force_auto_update` succeeds:
  - `config["forceautoupdate"] = force_auto_update` is executed
- the repo is marked with `forceautoupdate` set as `false`

`options[:force_auto_update]` does not exist:
- `force_auto_update` is set to `false`
- `unless force_auto_update` succeeds:
  - `config["forceautoupdate"] = force_auto_update` is executed
- the repo is marked with `forceautoupdate` set as `false` (**undesired outcome**)

Reply:
> will get blown away by some well-intentioned refactorer

There is a spec for it, but it could use a better dedcription than “defaults to nil”.

Reply:
Added a comment explaining why this defaults to `nil`.